packages/react/src/ui/Connect/components/Connector/Connecting.tsx

```diff
@@ -27,17 +27,13 @@ export function Connecting({ className }: ConnectorProps) {
     isConnected,
   } = useConnectUI();
 
-  if (!connector) return null;
+  useEffect(() => {
+    if (isConnected && route === Routes.CONNECTING && !isConnecting) {
```

Comment:
```suggestion
    if (isConnected && !isConnecting) {
```
this seems redundant since this component only renders for the Connecting route.